### DISCUSSION TITLE
fs_event_watch_dir test only creating files in the dir?

### DISCUSSION THREAD

**Discussion by hasansultan92** [libuv/libuv#4236] [Q&A] [answered]
Hi guys, hope you are doing well! I am looking to solve a long time issue in the file system with regards to FS events, focusing mainly on mac os. I am using the test case above to work with the kFSEvents from Apple and debugging libuv as I go.

I noticed that the test case is only creating files and then checking if the "watch" functionality is working fine? Should it not also be renaming files, deleting files, modifying files?

**Comment by bnoordhuis** [comment, marked as answer]
Maybe, but I think you'll find it's very hard to make file watcher tests run reliably on all platforms. They're already a perennial source of CI flakiness, even to this day.

**Reply by hasansultan92** [reply]
Interesting, can you give an example maybe of the CI Flakiness we have experienced thanks to file watchers?

**Reply by bnoordhuis** [reply]
E.g. https://github.com/libuv/libuv/issues/4106
